### PR TITLE
an idea for making it a module

### PR DESCRIPTION
**Pull request by mkuts12**
a better idea. what do you think?

### CODE REVIEW

**Comment by mkuts12** [comment]
it's based on what @corevo did in whiten

File: lib/index.js [hunk 1]
```diff
@@ -20,5 +20,14 @@ export function run(argv) {
   }
 
   var result = download(args);
-  tar.pack(result.path).pipe(fs.createWriteStream(args.join('-').replace(/:/g, '_') + '.tar')).on('end', result.delete);
+  tar.pack(result.path).pipe(fs.createWriteStream(args.join('-').replace(/:/g, '_') + '.tar')).on('end', () => {
```
Comment (Schniz) [inline lib/index.js]: Maybe you can save the `args.join('-').replace(/:/g, '_') + '.tar'` in a variable and return it with the callback?

File: lib/index.js [hunk 2]
```diff
@@ -20,5 +20,14 @@ export function run(argv) {
...
+      result.delete();
+      cb ? cb() : undefined;
```
Comment (Schniz) [inline lib/index.js]: `cb(tarFilePath)`?

File: lib/index.js (hunk 1 above)
Comment (mkuts12) [inline lib/index.js]: exactly what i thought after pushing it. already implemented the site with it

**Comment by mkuts12** [comment]
what about now?

**Comment by Schniz** [comment]
you changed everything :+1: 
is it good for merge?

**Comment by Schniz** [comment]
update the `README.md` and we'll merge it, yay!

**Comment by Schniz** [comment]
:star:

**Comment by mkuts12** [comment]
I use it to download in my site as a library. I would love you if you could check it for yourself. At least as a cli because it us practically the same.

**Comment by mkuts12** [comment]
Actually, i think i broke it yesterday what i started returning it as two different objects. 

What do you think i should do, return one object and make it clearer or stick to Tomer's standard?

**Comment by mkuts12** [comment]
*when

**Comment by Schniz** [comment]
I think that we should return it in `cb(err, data)` like node's standard.

@corevo what do you think? stick to node's `cb(err, data)` format in your lib too? :smile:

**Comment by mkuts12** [comment]
Done.

**Comment by mkuts12** [comment]
My friend.... can you accept the PR?

**Comment by Schniz** [comment]
Ill publish it soon

**Pull request merged**